srcbmi/mf6xmi.F90

```diff
@@ -97,6 +97,29 @@ module mf6xmi
 
 contains
 
+#if defined(__WITH_MPI__)
+  function xmi_initialize_mpi(mpi_comm) result(bmi_status) &
```

Comment:
Maybe in the future we think about putting this routine in a separate MPI labeled Fortran source file to keep MPI directives containerized?